sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php

```diff
@@ -1339,6 +1390,55 @@ function loop_status($lcd) {
 					$lcd_cmds[] = "widget_set {$name} title_wdgt 1 1 \"{$title}\"";
 					$lcd_cmds[] = "widget_set {$name} text_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"{$cputemperature}\"";
 					break;
+				case "scr_ntp":
+					$ntp_hms = date("H:i");
+					$ntpq_output = lcdproc_get_ntp_status();
+					if ($ntpq_output != false) {
+						$ntpq_counter = 0;
+						foreach ($ntpq_output as $line) {
+							if (substr($line, 0, 1) == "*" || substr($line, 0, 1) == "o") {
+								//Active NTP Peer
+								$line = substr($line, 1);
+								$peerinfo = preg_split("/[\s\t]+/", $line);
+								$peersrv = $peerinfo[0]; // Server IP
+								$peerref = $peerinfo[1]; // Ref ID
+								$peerstr = $peerinfo[2]; // Stratum
+								$peertyp = $peerinfo[3]; // Type
+								$peerdel = substr($peerinfo[7],0,5); // Delay
+								$peeroff = substr($peerinfo[8],0,6); // Offset
+								$peerjit = substr($peerinfo[9],0,5); // Jitter
+								// Common rows
+								$lcd_cmds[] = "widget_set {$name} time_st_wdgt 1 1 {$lcdpanel_width} 2 h 4 \"NTP: {$ntp_hms} Strtm: {$peerstr}\"";
+								$lcd_cmds[] = "widget_set {$name} text_wdgt 1 3 {$lcdpanel_width} 2 h 4 \"Delay Offset Jitter\"";
+								$lcd_cmds[] = "widget_set {$name} stats_wdgt 1 4 {$lcdpanel_width} 2 h 4 \"{$peerdel} {$peeroff} {$peerjit}\"";
+								if ($peertyp == "l" && $peerref == ".GPS.") {
+									// For local GPS, show the Ref ID of the source, and append how many satellites are in use or in view
+									$gps_sat_count = lcdproc_get_ntp_gps_sat_count();
+									$lcd_cmds[] = "widget_set {$name} ref_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"Src: {$peerref} " . (($gps_sat_count == false) ? "" : "{$gps_sat_count}") . "\"";
+								} elseif ($peertyp == "l") {
+									// For other local sources, show Ref ID of the source
+									$lcd_cmds[] = "widget_set {$name} ref_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"Src: {$peerref}\"";
+								} else {
+									// For everything else, show IP of the source
+									$lcd_cmds[] = "widget_set {$name} ref_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"Src: {$peersrv}\"";
+								}
+								$ntpq_counter++;
+							}
+						}
+					if ($ntpq_counter == 0) {
+						$lcd_cmds[] = "widget_set {$name} time_st_wdgt 1 1 {$lcdpanel_width} 2 h 4 \"NTP: No active peers\"";
+						$lcd_cmds[] = "widget_set {$name} ref_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"\"";
+						$lcd_cmds[] = "widget_set {$name} text_wdgt 1 3 {$lcdpanel_width} 2 h 4 \"\"";
+						$lcd_cmds[] = "widget_set {$name} stats_wdgt 1 4 {$lcdpanel_width} 2 h 4 \"\"";
```

Comment:
The indentation here seems wrong, it should be one level deeper. Otherwise it doesn't line up properly with the braces above/below

Reply:
in progress